### PR TITLE
fix(explore): Adhoc columns don't display correctly

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
Fix adhoc columns not displaying properly. When user added an adhoc column the label was blank and on click an empty popover was showing up instead of prefilled with custom sql.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
image

After:
image

### TESTING INSTRUCTIONS
1. Add an adhoc column
2. Verify that it displays correctly
3. Click on the column to edit
4. Verify that popover opens on Custom SQL tab with column's sql expression prefilled

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/packages/superset-ui-core/src/query/types/Column.ts [hunk 1]
```diff
@@ -61,7 +62,7 @@ export function isAdhocColumn(column?: any): column is AdhocColumn {
     typeof column !== 'string' &&
     column?.sqlExpression !== undefined &&
     column?.label !== undefined &&
-    column?.expressionType === 'SQL'
+    (!isDefined(column?.expressionType) || column?.expressionType === 'SQL')
```
Comment (zhaoyongjie) [inline superset-frontend/packages/superset-ui-core/src/query/types/Column.ts:65]: Maybe the same logic? 

```suggestion
....
  return (
    typeof column !== 'string' &&
    column?.label !== undefined &&
    (column?.expressionType === null || column?.expressionType === 'SQL')
```

File: superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndColumnSelect.test.tsx [hunk 2]
```diff
@@ -42,3 +44,19 @@ test('renders with value', () => {
...
   expect(screen.getByText('Column A')).toBeInTheDocument();
 });
+
+test('renders adhoc column', () => {
+  render(
+    <DndColumnSelect
+      {...defaultProps}
+      value={{
+        sqlExpression: 'Count *',
+        label: 'adhoc metric',
```
Comment (zhaoyongjie) [inline superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndColumnSelect.test.tsx:54]: A little fussy.....
```javascript
        sqlExpression: 'foo || bar',
        label: 'adhoc column',
```

File: superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndColumnSelect.test.tsx [hunk 3]
```diff
@@ -42,3 +44,19 @@ test('renders with value', () => {
...
+        expressionType: 'SQL',
+      }}
+    />,
+    { useDnd: true },
+  );
+  expect(screen.getByText('adhoc metric')).toBeVisible();
```
Comment (zhaoyongjie) [inline superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndColumnSelect.test.tsx:60]:
```suggestion
  expect(screen.getByText('adhoc column')).toBeVisible();
```

File: superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndColumnSelect.test.tsx (hunk 2 above)
Comment (kgabryje) [reply superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndColumnSelect.test.tsx:54]: Oops 😅

File: superset-frontend/packages/superset-ui-core/src/query/types/Column.ts (hunk 1 above)
Comment (kgabryje) [reply superset-frontend/packages/superset-ui-core/src/query/types/Column.ts:65]: I think we need to cover `undefined` here.  I'll change that to `column?.expressionType === undefined`

**Review by zhaoyongjie — APPROVED** [review]
looks good. Thanks for the fix.

**Comment by sadpandajoe** [comment]
🏷️ preset:2022.29

**Pull request merged**